meson.build

```diff
@@ -1268,6 +1268,8 @@ foreach ident : ['crypt_set_metadata_size',
         have_ident = have and cc.has_function(
                 ident,
                 prefix : '#include <libcryptsetup.h>',
+                # crypt_reencrypt() raises a deprecation warning so make sure -Wno-error is disabled.
+                args : '-Wno-error',
```

Comment:
Maybe, better to use `-Wno-deprecated-declarations` ??

```
Smaller alternative to https://github.com/systemd/systemd/pull/33165
that can go into v256.
```
The commit message is misleading, as this is necessary even if #33165 is merged.

Reply:
FWIW Passing `--werror` to `meson` without `-Werror` fixes that. `-Werror` interferes with the meson checks like this. I've just run into that when I passed `-Werror -Wp,-D_FORTIFY_SOURCE=3 -O2 -Werror`. `meson` added `-O0` to one of its check and it failed because `FORTIFY_SOURCE` throws warnings with `-O0`. It would be nice to switch `meson --werror` without the Werror kludge at some point now that https://github.com/mesonbuild/meson/issues/7360 is fixed.

Reply:
I have changed the PR to use --werror. I'm pretty sure your fix is not yet in all mesons of versions used by CI, but since we don't build with LTO, we might not see the issue in the first place, so let's see if it works. If not I'll revert back to -Werror